Stats: enable `stats/locations` feature flag on dev, stage and wpcalypso env

Initsogar opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Resolves https://github.com/Automattic/jetpack-roadmap/issues/2174

## Proposed Changes

* Enable `stats/locations` feature flag on development, stage and wpcalypso environments
  * This will also enable the new Locations module for all A12s and Calypso Live Branch.

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

Internal soft launch.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Spin up a Calypso Live Branch
* Ensure you see the new Locations module without the need of passing the explicit feature flag. 

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

Initsogar committed:
Enable stats/locations feature flag on development, stage and wpcalypso environments

dognose24 reviewed (approved):
LGTM!

kangzj commented:
It looks awesome! Two minor issues which are not blockers I'm pretty sure:

1. The gap grows when switching tabs:


https://github.com/user-attachments/assets/2c1a6e29-65fa-4e89-aa55-6897e6fbd531

2.  We might want to reduce the gap on the summary page:

<img width="1152" alt="Screenshot 2025-01-24 at 10 16 17 AM" src="https://github.com/user-attachments/assets/a27cdb8c-96c9-4840-8c24-21ecab475497" />


Great work team! Awesome to see this done so fast and beautifully 👍

myhro commented:
@kangzj Hi. My impression is that both issues will be fixed by #98767. Can you please review there as well when you have a chance?

Initsogar commented:
@kangzj thanks for your comments! As @myhro mentioned, we've just addressed those issues (and much more) in https://github.com/Automattic/wp-calypso/pull/98767.

We still have some adjustments to make, but I think we can go ahead and release this feature flag on staging. This way, it will be easier to walk through the feature and gather internal feedback without dealing with the Calypso live branch or development environments.
